.github/workflows/llvm.yml

```diff
@@ -9,18 +9,22 @@ on:
 jobs:
   llvm:
     runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        target:
+          - x86_64-unknown-linux-musl
     name: llvm
     outputs:
       cache-key: ${{ steps.cache-key.outputs.cache-key }}
     steps:
       - id: ls-remote
         run: |
           set -euxo pipefail
-          value=$(git ls-remote https://github.com/aya-rs/llvm-project.git refs/heads/rustc/19.1-2024-07-30 | cut -f1)
```

Comment:
let's leave this pointing to our repo? that way if we need to build from source (our sha doesn't match rust-lang's) we can do it without having to change this script.

alternatively we should couple this more strongly with the rustc version we're about to use (i.e. use the LLVM ref that our rustc was built with) but i'm not sure how to do that

Reply:
I left both SHAs now. And a comment why it's done.

(tl;dr Rust CI annotates LLVM tarballs with rust-lang/rust commits and rebuilds the llvm-project submodule for each commit)

Reply:
@tamird What I'm unsure about is whether we should use the Rust or aya/llvm commit hash in cache.

From one hand, following the rustc commit semms reasonable, because they might do some backports in their LLVM fork, which are going to land in Rust nightly, and this way we are just going to pick it up every time. The downside is that we will re-download LLVM every day.

If we go with aya/llvm, then we will most likely end up re-downloading LLVM only when we explicitly bump the branch. Saves resources, but requires us to update our fork if necessary and we might miss some updates. For now we usually sync the fork each half a year when Rust completely switches the major version of LLVM.

Reply:
I think neither of these is correct.

The way this is setup now you will download the LLVM associated with HEAD, not nightly. We should be using the sha we get from `rustc` (the binary), not the git repo. That would download a new LLVM almost every day, but on the same cadence as downloading a new rustc binary.